src/core/positioning/egenioussreceiver.cpp

```diff
@@ -0,0 +1,128 @@
+#include "egenioussreceiver.h"
+
+#include <QHostAddress>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+
+EgenioussReceiver::EgenioussReceiver( const QString &address, const int port, QObject *parent )
+  : AbstractGnssReceiver( parent ), mAddress( address ), mPort( port ), mTcpSocket( new QTcpSocket( this ) )
+{
+  setValid( !mAddress.isEmpty() && mPort > 0 );
+}
+
+void EgenioussReceiver::handleConnectDevice()
+{
+  connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
+  connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::onErrorOccurred );
+
+  if ( mAddress.isEmpty() || mPort == 0 )
+  {
+    return;
+  }
+
+  mTcpSocket->connectToHost( QHostAddress( mAddress ), mPort, QTcpSocket::ReadWrite );
+
+  if ( !mTcpSocket->waitForConnected( 3000 ) )
```

Comment:
Don't wait for connection, listen to signals to change connected status.

Reply:
Correct :+1: